packages/api/models/granules.js

```diff
@@ -576,60 +579,43 @@ class Granule extends Manager {
   }
 
   /**
-   * Parse a Cumulus message and build granule records for the embedded granules.
+   * Store a granule record.
    *
-   * @param {Object} cumulusMessage - A Cumulus message
-   * @returns {Promise<Array<Object>>} - An array of granule records
+   * @param {Object} granuleRecord - A granule record.
+   * @returns {Promise<Object|undefined>}
    */
-  static async _getGranuleRecordsFromCumulusMessage(cumulusMessage) {
-    const granules = getMessageGranules(cumulusMessage);
-    if (!granules) {
-      log.info(`No granules to process in the payload: ${JSON.stringify(cumulusMessage.payload)}`);
-      return [];
-    }
+  async _storeGranuleRecord(granuleRecord) {
+    const mutableFieldNames = this._getMutableFieldNames(granuleRecord);
+    const updateParams = this._buildDocClientUpdateParams({
+      item: granuleRecord,
+      itemKey: { granuleId: granuleRecord.granuleId },
+      mutableFieldNames,
+    });
 
-    const executionArn = getMessageExecutionArn(cumulusMessage);
-    const executionUrl = StepFunctionUtils.getExecutionUrl(executionArn);
+    // Only allow "running" granule to replace completed/failed
+    // granule if the execution has changed
+    if (granuleRecord.status === 'running') {
+      updateParams.ConditionExpression = '#execution <> :execution';
+    }
```

Comment:
What is a running granule?

Reply:
The sfEventSqsToDbRecords Lambda runs at the beginning and end of every Step Function, triggered by a Cloudwatch event. At the beginning of the workflow, its status is "running", so it saves the granules from the workflow as the database as being "running". This allows operators on the dashboard to see that the ingest of the granules is in process. Then when the workflow completes, another event fires to trigger the Lambda with a status of "completed", so the granules are re-saved as "completed".

Reply:
This document explains how data is "reported" from our workflows, though it only shows the data being saved to Dynamo:

https://nasa.github.io/cumulus/docs/data-cookbooks/ingest-notifications